gui/wxpython/startup/guiutils.py

```diff
@@ -244,9 +244,16 @@ def create_location_interactively(guiparent, grassdb):
             defineRegion.ShowModal()
             defineRegion.Destroy()
 
+        if gWizard.user_mapset:
+            user_mapset = create_mapset_interactively(guiparent,
+                                                      gWizard.grassdatabase,
+                                                      gWizard.location)
+            # Returns database and location created by user
+            # and a mapset user may want to switch to
+            gWizard_output = (gWizard.grassdatabase, gWizard.location,
+                              user_mapset)
```

Comment:
This `gWizard_output` value gets replaced by the following line (always). Play with return statements and variable in some trivial pure Python code example.

You need to define that fist as sort of a default and overwrite it here or just overwrite the last item (`gWizard_output[2] = ...)`)

Reply:
Addressed here:

if gWizard.user_mapset:
        mapset = create_mapset_interactively(guiparent,
                                                  gWizard.grassdatabase,
                                                  gWizard.location)
        # Returns database and location created by user
        # and a mapset user may want to switch to
        gWizard_output = (gWizard.grassdatabase, gWizard.location,
                          mapset)
    else:
        # Returns PERMANENT mapset when user mapset not defined
        gWizard_output = (gWizard.grassdatabase, gWizard.location,
                          "PERMANENT")